File: app/src/main/java/com/cazimir/relaxoo/ui/sound_grid/OnActivityCallback.java
package com.cazimir.relaxoo.ui.sound_grid;

import com.cazimir.relaxoo.dialog.OnDeleted;
import com.cazimir.relaxoo.model.Recording;
import com.cazimir.relaxoo.model.SavedCombo;
import com.cazimir.relaxoo.model.Sound;

import org.jetbrains.annotations.NotNull;

import java.util.HashMap;

public interface OnActivityCallback {
  void showNotification();

  void hideNotification();

  void showAddToFavoritesDialog(HashMap<Integer, Integer> playingSounds);

  void showToast(String message);

  void showTimerDialog();

  void triggerCombo(SavedCombo savedCombo);

  void showDeleteConfirmationDialog(OnDeleted recDeleted);

  void showBottomDialogForPro();

  void soundsFetchedAndSaved();

  void soundGridFragmentStarted();

  void hideSplash();

  void removeAds();

  void deleteRecording(@NotNull Recording recording);

  void renameRecording(@NotNull Recording recording, @NotNull String toString);

  void pinToDashBoardActionCalled(@NotNull Sound sound);
}

File: app/src/main/java/com/cazimir/relaxoo/ui/sound_grid/OnActivityCallback.kt
package com.cazimir.relaxoo.ui.sound_grid

import android.os.Bundle
import com.cazimir.relaxoo.dialog.OnDeleted
import com.cazimir.relaxoo.model.Recording
import com.cazimir.relaxoo.model.SavedCombo
import com.cazimir.relaxoo.model.Sound

interface OnActivityCallback {
    fun showAddToFavoritesDialog(sounds: List<Sound>)
    fun showTimerDialog()
    fun triggerCombo(savedCombo: SavedCombo)
    fun showDeleteConfirmationDialog(recDeleted: OnDeleted)
    fun showBottomDialogForPro()
    fun soundGridFragmentStarted()
    fun hideSplashScreen()
    fun removeAds()
    fun deleteRecording(recording: Recording)
    fun renameRecording(recording: Recording, toString: String)
    fun pinToDashBoardActionCalled(sound: Sound)
    fun playRewardAd()
    fun hideProgress()
    fun showMessageToUser(string: String, duration: Int)
    fun startBuyingProFlow()
    fun removeAdsViewAndButtonInAbout()
    fun logAnalyticsEvent(eventName: String, bundle: Bundle)
}